src/main/java/org/jabref/logic/integrity/UTF8Checker.java

```diff
@@ -0,0 +1,44 @@
+package org.jabref.logic.integrity;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+
+public class UTF8Checker implements EntryChecker {
+
+    /**
+     * Detect any non UTF-8 encoded field
+     */
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        List<IntegrityMessage> results = new ArrayList<>();
+        for (Map.Entry<Field, String> field : entry.getFieldMap().entrySet()) {
+            Charset charset = Charset.forName(System.getProperty("file.encoding"));
```

Comment:
I would extract this out of the loop, as it doesn't depend on the loop.

Reply:
What's the reason to use `System.getProperty("file.encoding")` and not say the encoding specified in the Library properties?

Reply:
Since different users have different charsets due to the operating system or the default settings of the computer. And `System.getProperty("file.encoding")` is used get the default charset. If the charset is not UTF-8, we should give a warning about that.
And the reason not to use the Library properties & Database properties: Maybe the user doesn't know the default charset in his computer or he set the charset for jabref, but we should give a warning about that since Non-UTF-8 charset may cause to garbled.

Reply:
Ok thanks for the explanation. 

But doesn't this give a lot of false positives? Say I have my library encoded in Charset A, and my systems default is Charset B. If all characters in my database are properly encoded with Charset A, then I shouldn't get any warnings even though some of the characters may not be encodable in Charset B, right?

But I also have to admit that I do not yet understand the use-case from the user perspective, so maybe I'm missing something obvious.

Reply:
I have thought about that. In my test, if one user's default charset is A then his paste-board, his input is all encoded in A.  So when he input something maybe just garbled. Maybe there is an example: https://github.com/JabRef/jabref/issues/7629
So the scenario may be rare. That's the reason I don't choose to get the charset by `Charset charset = bibDatabaseContext.getMetaData().getEncoding().orElse(preferences.getDefaultEncoding());` 

By the way, I have a question about the design. If the bibtex is only allowed in ascii in design, why do we allow the user to save it into different charsets?

Reply:
Hi, reviewer! @tobiasdiez  After thinking for a long time and doing some tests, I think maybe it's better to give 2 kinds of warning:
1. In BibLatex, if the Library charset is not UTF-8, then give a warning `Non-UTF-8 field found.`
2. In both BibLatex and BibTeX, if the System env is not UTF-8, give the warning `Non-UTF-8 env, may cause garbled.`
And I'm eagerly waiting for your suggestions and reply！